### PR TITLE
Modernize field access in yt scripts

### PR DESCRIPTION
**Pull request by lucafedeli88**
As suggested by @ax3l  in https://github.com/ECP-WarpX/WarpX/pull/2454, this PR modernizes field access in `yt` analysis scripts by replacing `ad["FIELD"]` with `ad["mesh","FIELD"]` (or `ad["boxlib", "FIELD"]` ). This is done for several analysis scripts.

I have to confess that I don't know when I should use "mesh" instead of "boxlib" ...

X-ref: #2117

### CODE REVIEW

**Comment by ax3l** [comment]
> I have to confess that I don't know when I should use "mesh" instead of "boxlib" ...

I think mesh for fields and boxlib for particles... @atmyers might know more :)

**Comment by atmyers** [comment]
> > I have to confess that I don't know when I should use "mesh" instead of "boxlib" ...
> 
> I think mesh for fields and boxlib for particles... @atmyers might know more :)

Field data can be referenced either by "mesh" or "boxlib" - they should be fully equivalent. I'd prefer "mesh". Particles are referenced like `(<species_name>, <component_name>)`, for example `("beam", "particle_id")`. Or you can do `("all", "particle_weight")` which will return a big array with all species concatenated together.

File: Examples/Tests/Langmuir/analysis_langmuir_multi_rz.py [hunk 1]
```diff
@@ -119,8 +119,8 @@ def Ez( z, r, epsilon, k0, w0, wp, t) :
 
 # Check charge conservation (relative L-infinity norm of error) with current correction
 if current_correction:
-    divE = data['divE'].to_ndarray()
-    rho  = data['rho' ].to_ndarray() / epsilon_0
+    divE = data[('boxlib','divE')].to_ndarray()
+    rho  = data[('boxlib','rho')].to_ndarray() / epsilon_0
```
Comment (lucafedeli88) [inline Examples/Tests/Langmuir/analysis_langmuir_multi_rz.py:123]:
```suggestion
    divE = data[('mesh','divE')].to_ndarray()
    rho  = data[('mesh','rho')].to_ndarray() / epsilon_0
```

File: Examples/Tests/Langmuir/analysis_langmuir_multi.py [hunk 2]
```diff
@@ -133,8 +133,8 @@ def get_theoretical_field( field, t ):
 # current correction (psatd.do_current_correction=1) is applied or when
 # Vay current deposition (algo.current_deposition=vay) is used
 if current_correction or vay_deposition:
-    rho  = data['rho' ].to_ndarray()
-    divE = data['divE'].to_ndarray()
+    rho  = data[('boxlib','rho')].to_ndarray()
+    divE = data[('boxlib','divE')].to_ndarray()
```
Comment (lucafedeli88) [inline Examples/Tests/Langmuir/analysis_langmuir_multi.py:137]:
```suggestion
    rho  = data[('mesh','rho')].to_ndarray()
    divE = data[('mesh','divE')].to_ndarray()
```

File: Examples/Tests/Langmuir/analysis_langmuir_multi_2d.py [hunk 3]
```diff
@@ -109,8 +109,8 @@ def get_theoretical_field( field, t ):
 # current correction (psatd.do_current_correction=1) is applied or when
 # Vay current deposition (algo.current_deposition=vay) is used
 if current_correction or vay_deposition:
-    rho  = data['rho' ].to_ndarray()
-    divE = data['divE'].to_ndarray()
+    rho  = data[('boxlib','rho')].to_ndarray()
+    divE = data[('boxlib','divE')].to_ndarray()
```
Comment (lucafedeli88) [inline Examples/Tests/Langmuir/analysis_langmuir_multi_2d.py:113]:
```suggestion
    rho  = data[('mesh','rho')].to_ndarray()
    divE = data[('mesh','divE')].to_ndarray()
```

File: Examples/Tests/Langmuir/analysis_langmuir_multi.py [hunk 4]
```diff
@@ -133,8 +133,8 @@ def get_theoretical_field( field, t ):
 # current correction (psatd.do_current_correction=1) is applied or when
 # Vay current deposition (algo.current_deposition=vay) is used
 if current_correction or vay_deposition:
-    rho  = data['rho' ].to_ndarray()
-    divE = data['divE'].to_ndarray()
+    rho  = data[('mesh','rho')].to_ndarray()
```
Comment (lucafedeli88) [inline Examples/Tests/Langmuir/analysis_langmuir_multi.py]:
```suggestion
    rho  = data[('mesh','boxlib')].to_ndarray()
```

File: Examples/Tests/Langmuir/analysis_langmuir_multi_2d.py [hunk 5]
```diff
@@ -109,8 +109,8 @@ def get_theoretical_field( field, t ):
 # current correction (psatd.do_current_correction=1) is applied or when
 # Vay current deposition (algo.current_deposition=vay) is used
 if current_correction or vay_deposition:
-    rho  = data['rho' ].to_ndarray()
-    divE = data['divE'].to_ndarray()
+    rho  = data[('mesh','rho')].to_ndarray()
```
Comment (lucafedeli88) [inline Examples/Tests/Langmuir/analysis_langmuir_multi_2d.py]:
```suggestion
    rho  = data[('boxlib','rho')].to_ndarray()
```

File: Examples/Tests/Langmuir/analysis_langmuir_multi_rz.py [hunk 6]
```diff
@@ -119,8 +119,8 @@ def Ez( z, r, epsilon, k0, w0, wp, t) :
 
 # Check charge conservation (relative L-infinity norm of error) with current correction
 if current_correction:
-    divE = data['divE'].to_ndarray()
-    rho  = data['rho' ].to_ndarray() / epsilon_0
+    divE = data[('mesh','divE')].to_ndarray()
+    rho  = data[('mesh','rho')].to_ndarray() / epsilon_0
```
Comment (lucafedeli88) [inline Examples/Tests/Langmuir/analysis_langmuir_multi_rz.py]:
```suggestion
    rho  = data[('boxlib','rho')].to_ndarray() / epsilon_0
```

File: Examples/Tests/Langmuir/analysis_langmuir_multi.py [hunk 7]
```diff
@@ -133,8 +133,8 @@ def get_theoretical_field( field, t ):
 # current correction (psatd.do_current_correction=1) is applied or when
 # Vay current deposition (algo.current_deposition=vay) is used
 if current_correction or vay_deposition:
-    rho  = data['rho' ].to_ndarray()
-    divE = data['divE'].to_ndarray()
+    rho  = data[('mesh','boxlib')].to_ndarray()
+    divE = data[('mesh','divE')].to_ndarray()
```
Comment (lucafedeli88) [inline Examples/Tests/Langmuir/analysis_langmuir_multi.py]:
```suggestion
    rho  = data[('boxlib','rho')].to_ndarray()
    divE = data[('boxlib,'divE')].to_ndarray()
```

File: Examples/Tests/Langmuir/analysis_langmuir_multi_2d.py [hunk 8]
```diff
@@ -109,8 +109,8 @@ def get_theoretical_field( field, t ):
 # current correction (psatd.do_current_correction=1) is applied or when
 # Vay current deposition (algo.current_deposition=vay) is used
 if current_correction or vay_deposition:
-    rho  = data['rho' ].to_ndarray()
-    divE = data['divE'].to_ndarray()
+    rho  = data[('boxlib','rho')].to_ndarray()
+    divE = data[('mesh','divE')].to_ndarray()
```
Comment (lucafedeli88) [inline Examples/Tests/Langmuir/analysis_langmuir_multi_2d.py]:
```suggestion
    rho  = data[('boxlib','rho')].to_ndarray()
    divE = data[('boxlib','divE')].to_ndarray()
```

File: Examples/Tests/Langmuir/analysis_langmuir_multi_rz.py [hunk 9]
```diff
@@ -119,8 +119,8 @@ def Ez( z, r, epsilon, k0, w0, wp, t) :
 
 # Check charge conservation (relative L-infinity norm of error) with current correction
 if current_correction:
-    divE = data['divE'].to_ndarray()
-    rho  = data['rho' ].to_ndarray() / epsilon_0
+    divE = data[('mesh','divE')].to_ndarray()
+    rho  = data[('boxlib','rho')].to_ndarray() / epsilon_0
```
Comment (lucafedeli88) [inline Examples/Tests/Langmuir/analysis_langmuir_multi_rz.py]:
```suggestion
    divE = data[('boxlib','divE')].to_ndarray()
    rho  = data[('boxlib','rho')].to_ndarray() / epsilon_0
```

File: Examples/Tests/Langmuir/analysis_langmuir_multi.py [hunk 10]
```diff
@@ -133,8 +133,8 @@ def get_theoretical_field( field, t ):
 # current correction (psatd.do_current_correction=1) is applied or when
 # Vay current deposition (algo.current_deposition=vay) is used
 if current_correction or vay_deposition:
-    rho  = data['rho' ].to_ndarray()
-    divE = data['divE'].to_ndarray()
+    rho  = data[('boxlib','rho')].to_ndarray()
+    divE = data[('boxlib,'divE')].to_ndarray()
```
Comment (lucafedeli88) [inline Examples/Tests/Langmuir/analysis_langmuir_multi.py]:
```suggestion
    divE = data[('boxlib','divE')].to_ndarray()
```

File: Examples/Modules/boosted_diags/analysis_3Dbacktransformed_diag.py [hunk 11]
```diff
@@ -23,7 +23,7 @@
 snapshot = './lab_frame_data/snapshots/snapshot00002'
 header   = './lab_frame_data/snapshots/Header'
 allrd, info = read_raw_data.read_lab_snapshot(snapshot, header)
-F = allrd['Ez']
+F = allrd[('mesh','Ez')]
```
Comment (lucafedeli88) [inline Examples/Modules/boosted_diags/analysis_3Dbacktransformed_diag.py]:
```suggestion
F = allrd['Ez']
```

File: Examples/Modules/boosted_diags/analysis_3Dbacktransformed_diag.py [hunk 12]
```diff
@@ -32,7 +32,7 @@
 snapshot_slice = './lab_frame_data/slices/slice00002'
 header_slice   = './lab_frame_data/slices/Header'
 allrd, info = read_raw_data.read_lab_snapshot(snapshot_slice, header_slice)
-Fs = allrd['Ez']
+Fs = allrd[('mesh','Ez')]
```
Comment (lucafedeli88) [inline Examples/Modules/boosted_diags/analysis_3Dbacktransformed_diag.py]:
```suggestion
Fs = allrd['Ez']
```

**Comment by ax3l** [comment]
re-opening to trigger CI

**Review by ax3l — APPROVED** [review]
Thank you! :tada:

**Pull request merged**